### PR TITLE
feat(SetTheory/Ordinal/Arithmetic): generalize Ordinal.lt_iSup

### PR DESCRIPTION
**Pull request by YnirPaz**
Generalize the universes of `Ordinal.lt_iSup`.

---

`Logic.UnivLE` contains the `Small.{max u v, u}` instance, so it needs to be imported in Ordinal/FixedPoint.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Ordinal/Arithmetic.lean [hunk 1]
```diff
@@ -1133,9 +1159,9 @@ theorem le_sup {ι : Type u} (f : ι → Ordinal.{max u v}) : ∀ i, f i ≤ sup
   Ordinal.le_iSup f i
 
 /-- `ciSup_le_iff'` whenever the outputs live in a higher universe than the inputs. -/
-protected theorem iSup_le_iff {ι : Type u} {f : ι → Ordinal.{max u v}} {a} :
+protected theorem iSup_le_iff {ι} {f : ι → Ordinal.{u}} {a : Ordinal.{u}} [Small.{u} ι] :
     iSup f ≤ a ↔ ∀ i, f i ≤ a :=
-  ciSup_le_iff' (bddAbove_range f)
+  ciSup_le_iff' (bddAbove_iff_small.mpr (small_range f))
```
Comment (vihdzp) [inline Mathlib/SetTheory/Ordinal/Arithmetic.lean:1164]: Does this work?
```suggestion
  ciSup_le_iff' (bddAbove_of_small _)
```

File: Mathlib/SetTheory/Ordinal/Arithmetic.lean [hunk 2]
```diff
@@ -1133,9 +1159,9 @@ theorem le_sup {ι : Type u} (f : ι → Ordinal.{max u v}) : ∀ i, f i ≤ sup
   Ordinal.le_iSup f i
 
 /-- `ciSup_le_iff'` whenever the outputs live in a higher universe than the inputs. -/
-protected theorem iSup_le_iff {ι : Type u} {f : ι → Ordinal.{max u v}} {a} :
+protected theorem iSup_le_iff {ι} {f : ι → Ordinal.{u}} {a : Ordinal.{u}} [Small.{u} ι] :
```
Comment (vihdzp) [inline Mathlib/SetTheory/Ordinal/Arithmetic.lean:1162]: Can't you generalize `le_iSup`, `iSup_le_iff`, and `iSup_le` in the exact same way? That would ensure consistency across these four functions.

File: Mathlib/SetTheory/Ordinal/FixedPoint.lean [hunk 3]
```diff
@@ -5,6 +5,7 @@ Authors: Violeta Hernández Palacios, Mario Carneiro
 -/
 import Mathlib.SetTheory.Ordinal.Arithmetic
 import Mathlib.SetTheory.Ordinal.Exponential
+import Mathlib.Logic.UnivLE
```
Comment (vihdzp) [inline Mathlib/SetTheory/Ordinal/FixedPoint.lean:8]: Wait, the `UnivLE` API can be used to construct a `Small.{max u v, u}` instance? I was under the impression this just didn't work, if it does then that's awesome!

File: Mathlib/SetTheory/Ordinal/Arithmetic.lean (hunk 2 above)
Comment (YnirPaz) [reply Mathlib/SetTheory/Ordinal/Arithmetic.lean:1162]: `le_iSup` became a bit worse at filling in the function argument which is unfortunate, and besides that there are a bunch of similarly generalizable lemmas in this file that I'll deal with in a seperate PR (tomorrow if I have time).

Comment (YnirPaz) [reply Mathlib/SetTheory/Ordinal/Arithmetic.lean:1162]: #16996

Comment (vihdzp) [reply Mathlib/SetTheory/Ordinal/Arithmetic.lean:1162]: I think it'd be good to have all this as a single PR. You can close this one and keep the other one open as is.

**Pull request closed without merging**